net/rhp/contracts.go

```diff
@@ -139,100 +138,120 @@ func ValidateContractRenewal(existing, renewal types.FileContract, currentHeight
 	return nil
 }
 
-func validateStdRevision(vc consensus.ValidationContext, current, revision Contract) error {
+// ValidateContractFinalization verifies that the revision locks the current
+// contract by setting its revision number to the maximum legal value, no other
+// fields should change. Signatures are not validated.
+func ValidateContractFinalization(current, final types.FileContract) error {
```

Comment:
could this be implemented like so?
```go
func ValidateContractFinalization(current, final types.FileContract) error {
    exp := current
    exp.RevisionNumber = types.MaxRevisionNumber
    if exp != final {
        return errors.New("finalized contract changes fields other than the revision number")
    }
    return nil
}
```

Reply:
The signatures need to be zeroed out and a generic "something changed" error is a little less friendly, but it would work

Reply:
true, it's a bit less friendly. On the other hand, it's future-proof against changes to the `FileContract` type.

idk, it's probably fine either way.